### PR TITLE
Flash buff

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

fixes area flash being broken.
makes melee flash hits stun for 1.5 seconds

## Why / Balance

Melee flashes are pretty garbage. It's basically useless compared to just using the area flash. Not being able to properly stun people also makes flashes pretty worthless as a limited control tool (for people like heads).

 Making the flash actually stun allows them to be a lot more useful for defense while being much more counterable than a baton due to the prevalence of flash protection. This makes heads way more able to deal with shitters via using flashes. it also reduces a pretty common metagaming aspect of revs.

## Technical details

the predicate for ignoring on the raycast was brokey.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by Plykiya** [comment]
Science antags were already the greatest power gamers, I am ready for them to become even more powerful than before!

**Comment by Brandon-Huu** [comment]
Wait what?
Like stun stun? On the ground stun?

**Comment by vorkathbruh** [comment]
the stun should not last long enough to cuff someone, which i think it barely does at 3 seconds

**Comment by Brandon-Huu** [comment]
> the stun should not last long enough to cuff someone, which i think it barely does at 3 seconds

yeah but still, you can just stun them with the flash then bash their head in with the baton

**Comment by thebadman4662** [comment]
Does this stun prevent stamcrit or its something like shock stun?

**Comment by Holinka4ever** [comment]
As people told before, anyone can use flashing device as compact stunbaton, without usinh stunbatom.
Its good buff for flash, but it will turn station into epileptic cautious location.
So what about to give stun property to flashbangs types? It will save people's nervous from tiders that break into robotics.

**Comment by thebadman4662** [comment]
That would mean stingers stun you even if they do not stamstun you... :trollface:

**Comment by northerndakotas** [comment]
An instant stun/disarm against anyone not wearing sunglasses feels, at first glance, like it might end up being not awesome.

Perhaps it would be better if the melee flashing function simply blinded for longer? Long enough to escape or otherwise beat the person into submission without taking away any chance for them to fight back.

Lastly, I thought flashes were already reduced in volume to cut down on rev metagaming. If they weren't, that would probably be the better change to reduce flash callouts.

**Comment by thebadman4662** [comment]
Melee flash already blinds and slows for 300% duration. Which is plenty to stunbaton.

**Comment by northerndakotas** [comment]
> Melee flash including wideswing already blinds and slows for 300% duration, which is plenty to stunbaton.

I see, if that's the case I don't think I could support a buff for flashes right now. At least, not one that will disarm anyone at will with a single swing.

**Comment by SpeltIncorrectyl** [comment]
Can we give all heads of staff sunglasses?

**Comment by IlyaElDunaev** [comment]
Antagonists have no chance of winning in close combat

**Comment by deltanedas** [comment]
this seems way too strong and just to allow headrevs to flash people in the bar openly and not instantly be valid

**Comment by asperger-sind** [comment]
It really is way too strong; flashes as-is are 90% of the time a "hey, you win this combat interaction now" tool, making them also stun you on-hit without nerfing the AOE range is really overpowered. There are about 10 people on a fully filled highpop station with flashres. You still can win combat interactions against the majority of the station just by having a flash right now and this doesn't make it better.

This also effectively forces antagonists (as if they are not already forced) into buying or acquiring flash resistance if they intend to interact with sec/command in any way at all (which makes up the bulk of traitor objectives) to avoid getting trolled by an instant on-demand 5 charge stun that also has an AOE blind feature.

Flashes should put you into a disadvantage if you're fighting against them without flashres but shouldn't make you instantly lose the fight.

I see a different solution to the problem behind this, melee flashes should temporarily allow you to forcefully grab someone and (if it does not already) should also just have a longer blind and slow duration, during the force-grab you can just drag them into a disposal bin or out of your dept, PJB doesn't have to chloral/crit Urist to get them out of medbay and everyone is happy.

It still is a flash buff (so, not very pretty balance-wise imo) but now it doesn't upset general combat balance as much, this could also be turned to a sidegrade by nerfing AoE flash range or blind/slow duration, which is pretty good anyway because this is the flashbang use case.

**Comment by Plykiya** [comment]
> Can we give all heads of staff sunglasses?

NanoTrasen contact lenses, all command members immune to flashes

**Comment by K-Dynamic** [comment]
> NanoTrasen contact lenses, all command members immune to flashes!!..... I wish.

Maybe make flashproof implants or give flash immunity to mindshield implants?

**Comment by thebadman4662** [comment]
Guys, syndicate flashbang kit is already on life support, do not kill it with even more command buffs. :(

**Comment by XxSWAG-MASTER-OVERLORDxX** [comment]
> Guys, syndicate flashbang kit is already on life support, do not kill it with even more command buffs. :(

while this is a sec and command buff, this is an INSANE revs buff and im 100% for it

**Comment by thebadman4662** [comment]
> > Guys, syndicate flashbang kit is already on life support, do not kill it with even more command buffs. :(
> 
> while this is a sec and command buff, this is an INSANE revs buff and im 100% for it

How command and sec being immune to flash effects 100% of the time without sec-/sunglassess is a buff to revs again? It also means all of sec is running sechud all the time and everyone wants mindshield regardless of gamemode if they are not rev.

**Comment by TurboTrackerss14** [comment]
While I have my own personal bias for disliking this pr in general.

I want to voice a concern from myself (and probably other admins) regarding how this will affect revolutionary identification 
In our rule clarification surrounding rev identification, dropping to the floor after a flash is a direct mention of how to suspect revolutionaries. https://forum.spacestation14.io/index.php?/topic/3450-revealing-a-revolution

While this is going to need admin team discussion also I wanted to reference it here as it has a knock on effect on how security/command can identify/suspect revs.

**Comment by EmoGarbage404** [comment]
> I want to voice a concern from myself (and probably other admins) regarding how this will affect revolutionary identification In our rule clarification surrounding rev identification, dropping to the floor after a flash is a direct mention of how to suspect revolutionaries. https://forum.spacestation14.io/index.php?/topic/3450-revealing-a-revolution

to be quite honest, the fact that the malformed porting of flashes from SS13 was accepted as an intended feature rather than an inconsistency to remedy makes the ruling listed extremely comical. IMO, it should just be removed entirely as a method of suspect.

File: Content.Shared/Flash/Components/FlashComponent.cs [hunk 1]
```diff
@@ -12,6 +12,18 @@ public sealed partial class FlashComponent : Component
...
+        /// </summary>
+        [DataField]
+        public TimeSpan MeleeStunDuration = TimeSpan.FromSeconds(1.5);
+
+        /// <summary>
+        /// If true, melee hits will do a hard stun.
+        /// </summary>
+        [DataField]
+        public bool StunOnMelee = true;
+
```
Comment (metalgearsloth) [inline Content.Shared/Flash/Components/FlashComponent.cs:26]: Make the stun duration nullable and you can reduce it to 1 field.

**Comment by SoulFN** [comment]
Maybe 0,5 second would be balanced?

**Pull request merged**